### PR TITLE
Implementation of `clz`, `ctz`, `popcnt`

### LINKED ISSUE #11: Clear TODOs in semantics/test-set
As the semantics is being built, several TODO statements are left in place to remind us of things that need to be done.

These should be fairly mechanical to implement.

### PR DESCRIPTION
**Pull request by hjorthjort**
Fix #11.

### CODE REVIEW

File: wasm.md [hunk 1]
```diff
@@ -197,7 +197,35 @@ The rotation operators `rotl` and `rotr` do not have appropriate K builtins, and
...
+    rule #popcnt(N) => #bool(N modInt 2 ==Int 1) +Int #popcnt(N >>Int 1)
+```
+
+`clz` counts the number of leading zero-bits, with 0 having all leading zero-bits.
+`ctz` counts the number of trailing zero-bits, with 0 having all trailing zero-bits.
+`popcnt` counts the number of non-zero bits.
+
+```
+    syntax IUnOp ::= "clz" | "ctz" | "popcnt"
+ // ----------------------
```
Comment (ehildenb) [inline wasm.md]: Lets make sure the `---` underneath extends all the way to the end of the `syntax` declaration.

File: wasm.md [hunk 2]
```diff
@@ -197,7 +197,35 @@ The rotation operators `rotl` and `rotr` do not have appropriate K builtins, and
...
+                 | #popcnt   ( Int ) [function]
+    // ----------------------------------------
+    rule #minWidth(0) => 0
+    rule #minWidth(N) => 1 +Int #minWidth(N >>Int 1)requires N =/=Int 0
+
+    rule #ctz(0) => 0
+    rule #ctz(N) => #if N modInt 2 ==Int 1 #then 0 #else 1 +Int #ctz(N >>Int 1) #fi
+
+    rule #popcnt(0) => 0
+    rule #popcnt(N) => #bool(N modInt 2 ==Int 1) +Int #popcnt(N >>Int 1)
```
Comment (ehildenb) [inline wasm.md]: These rules overlap on the `N == 0` case, which will cause case-splitting during symbolic execution. Let's add a `requires N =/=Int 0` case to the second of each rule.

Comment (hjorthjort) [reply wasm.md]: Thanks, good catch.

File: wasm.md [hunk 3]
```diff
@@ -197,7 +197,35 @@ The rotation operators `rotl` and `rotr` do not have appropriate K builtins, and
...
 
-**TODO**: `clz`, `ctz`, `popcnt`.
+The bit counting operators also lack appropriate K builtins, and are implemented by using width-agnostic helper functions.
+Note: The actual `ctz` operator considers the integer 0 to have *all* zero-bits, whereas the `#ctz` helper function considers it to have *no* zero-bits, in order for it to be width-agnostic.
+
+```k
+    syntax Int ::= #minWidth ( Int ) [function]
+                 | #ctz      ( Int ) [function]
+                 | #popcnt   ( Int ) [function]
+    // ----------------------------------------
```
[the remark this answers is not in the document]
Comment (hjorthjort) [reply wasm.md]: Rats. Thanks for the catch.

**Pull request merged**